ovirt_imageio/client/_ovirt.py

```diff
@@ -0,0 +1,378 @@
+            transfer. If not specified engine will pick a random host.
+        backup (ovirtsdk4.types.Backup): When downloading backup, the
+            backup object owning the disks.
+        inactivity_timeout (int): Number of seconds engine will wait for
+            client activity before pausing the transfer. If not set, use
+            engine default value.
+        timeout (float, optional): number of seconds to wait for transfer
+            to become ready.
+        shallow (bool): Download only the specified image instead of the
+            entire image chain. When downloading a disk transfer only
+            the active disk snapshot data. When downloading a disk
+            snapshot, transfer only the specified disk snaphost data.
+        timeout_policy (ovirtsdk4.types.ImageTransferTimeoutPolicy): the
+            action to take after inactivity timeout.
+
+    Returns:
+        ovirtsdk4.types.ImageTransfer in phase TRANSFERRING
+    """
+    log.info(
+        "Creating transfer image=%s direction=%s host=%s backup=%s "
+        "shallow=%s timeout_policy=%s",
+        Repr(image, "id"),
+        direction,
+        Repr(host, "name"),
+        Repr(backup, "id"),
+        shallow,
+        timeout_policy)
+
+    start = time.monotonic()
+    deadline = start + timeout
+
+    transfer = types.ImageTransfer(
+        host=host,
+        direction=direction,
+        backup=backup,
+        inactivity_timeout=inactivity_timeout,
+        timeout_policy=timeout_policy,
+        format=types.DiskFormat.RAW,
+        shallow=shallow)
+
+    if isinstance(image, types.Disk):
+        transfer.disk = image
+    elif isinstance(image, types.DiskSnapshot):
+        transfer.snapshot = image
+    else:
+        raise ValueError(f"Cannot transfer {Repr(image)}")
+
+    transfers_service = con.system_service().image_transfers_service()
+    transfer = transfers_service.add(transfer)
+
+    # At this point the transfer owns the disk and will delete the disk
+    # if the transfer is canceled, or if finalizing the transfer fails.
+
+    transfer_service = _transfer_service(con, transfer.id)
+
+    while True:
+        try:
+            transfer = transfer_service.get()
+        except sdk.NotFoundError:
+            # The system has removed the disk and the transfer.
+            raise RuntimeError(f"Transfer {transfer.id} was removed")
+
+        if transfer.phase == types.ImageTransferPhase.FINISHED_FAILURE:
+            # The system will remove the disk and the transfer soon.
+            raise RuntimeError(f"Transfer {transfer.id} has failed")
+
+        if transfer.phase == types.ImageTransferPhase.PAUSED_SYSTEM:
+            transfer_service.cancel()
+            raise RuntimeError(f"Transfer {transfer.id} was paused by system")
+
+        if transfer.phase == types.ImageTransferPhase.TRANSFERRING:
+            break
+
+        if transfer.phase != types.ImageTransferPhase.INITIALIZING:
```

Comment:
The PAUSED_* phases are not valid but are a reason to cancel?

Reply:
Yes, if the system paused the transfer something is wrong and we cannot continue.
The best way is to cancel the transfer and fail.